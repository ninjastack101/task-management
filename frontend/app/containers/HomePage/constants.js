export const FETCH_TASKS = 'boilerplate/Home/FETCH_TASKS';
export const LOAD_TASKS = 'boilerplate/Home/LOAD_TASKS';
export const ADD_TASK = 'boilerplate/Home/ADD_TASK';
export const EDIT_TASK = 'boilerplate/Home/EDIT_TASK';
export const DELETE_TASK = 'boilerplate/Home/DELETE_TASK';
