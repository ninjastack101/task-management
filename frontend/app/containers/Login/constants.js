export const CALL_SIGNUP = "boilerplate/LOGIN/CALL_SIGNUP";
export const SIGNUP_SUCCESS = "boilerplate/LOGIN/SIGNUP_SUCCESS";
export const LOGOUT = "boilerplate/LOGIN/LOGOUT";
