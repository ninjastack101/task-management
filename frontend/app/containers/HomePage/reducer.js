import produce from 'immer';
import { LOAD_TASKS } from './constants';

// The initial state of the App
export const initialState = {
  tasks: [],
};

/* eslint-disable default-case, no-param-reassign */
const homeReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_TASKS:
        draft.tasks = action.tasks;
        break;
    }
  });

export default homeReducer;
