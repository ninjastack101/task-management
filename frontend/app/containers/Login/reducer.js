import produce from "immer";
import { SIGNUP_SUCCESS, LOGOUT } from "./constants";

// The initial state of the App
export const initialState = {
  username: "",
  userId: ""
};

/* eslint-disable default-case, no-param-reassign */
const loginReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case SIGNUP_SUCCESS:
        draft.userId = action.id;
        draft.username = action.username;
        break;
      case LOGOUT:
        draft.userId = "";
        draft.username = "";
        break;
    }
  });

export default loginReducer;
