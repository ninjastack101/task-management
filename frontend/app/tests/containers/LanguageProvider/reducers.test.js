import reducer from '../../../containers/LanguageProvider/reducer';
import { CHANGE_LOCALE } from '../../../containers/LanguageProvider/constants';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('Fetch home reducer', () => {

  let initialState = {
    locale: DEFAULT_LOCALE
  };

  it('Should return the initial state', (done) => {
    expect(reducer(undefined, {})).toEqual(initialState);
    done();
  });

  it('should handle CHANGE_LOCALE', (done) => {

    const startAction = {
      type: CHANGE_LOCALE,
      locale: {
        0: "somestring"
      }
    };
    expect(reducer(initialState, startAction)).toMatchSnapshot();
    done();
  });
});


