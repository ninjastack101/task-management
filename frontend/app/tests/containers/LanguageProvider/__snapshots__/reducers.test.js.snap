// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch home reducer should handle CHANGE_LOCALE 1`] = `
Object {
  "locale": Object {
    "0": "somestring",
  },
}
`;
